syntax = "proto2";

package AuthProtocol;

message CreateAccountWeb
{
    required int64 requestID = 1;
    required string email = 2;
    required string passwd = 3;
}

message CreateAccountWebSuccess
{
    required int64 requestID = 1;
    required int64 userID = 2;
}

message CreateAccountWebFailure 
{
    required int64 requestId = 1;
    enum reason 
    {
        ACCOUNT_ALREADY_EXISTS = 0;
        INVALID_PASSWORD = 1;
        INTERNAL_SERVER_ERROR = 2;
    }
    required reason Fail_reason = 2;
}

message AuthenticateWeb 
{
    required int64 requestId = 1;
    required string email = 2;
    required string passwd = 3;
}

message AuthenticateWebSuccess {
    required int64 requestId = 1;
    required int64 userId = 2;
    required string creationDate = 3;
}

message AuthenticateWebFailure {
    required int64 requestId = 1;
    enum reason 
    {
        INVALID_CREDENTIALS = 0;
        INTERNAL_SERVER_ERROR = 1;
    }
    required reason Fail_reason = 2;
}

message Request {
    required int64 requestId = 1;
    enum CMD_TYPE
    {
        CREATE_ACCOUNT = 0;
        AUTHENTICATE = 1;
    }
    required CMD_TYPE cmd = 2;
    optional CreateAccountWeb createAcc = 3;
    optional AuthenticateWeb authen = 4;
}

message Response {
    required int64 requestId = 1;
    enum Res_TYPE
    {
        SUCCESS = 0;
        INVALID_CREDENTIALS = 1;
        ACCOUNT_EXISTS = 2;
        INVALID_PASS = 3;
        INTERNAL_SERVER_ERROR = 4;
    }
    required Res_TYPE resp_msg_type = 2;
}
    